platforms/ide/problems/src/main/kotlin/org/gradle/problems/internal/DefaultProblemsReportCreator.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.problems.internal
+
+import org.gradle.api.internal.DocumentationRegistry
+import org.gradle.api.internal.StartParameterInternal
+import org.gradle.api.internal.file.temp.TemporaryFileProvider
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.problems.internal.FileLocation
+import org.gradle.api.problems.internal.Problem
+import org.gradle.internal.buildoption.InternalOptions
+import org.gradle.internal.cc.impl.problems.BuildNameProvider
+import org.gradle.internal.cc.impl.problems.JsonSource
+import org.gradle.internal.cc.impl.problems.JsonWriter
+import org.gradle.internal.concurrent.ExecutorFactory
+import org.gradle.internal.configuration.problems.CommonReport
+import org.gradle.internal.configuration.problems.FailureDecorator
+import org.gradle.internal.configuration.problems.StructuredMessage
+import org.gradle.internal.configuration.problems.writeError
+import org.gradle.internal.configuration.problems.writeStructuredMessage
+import org.gradle.internal.logging.ConsoleRenderer
+import org.gradle.internal.operations.OperationIdentifier
+import org.gradle.internal.problems.failure.FailureFactory
+import org.gradle.problems.buildtree.ProblemReporter
+import java.io.File
+
+val logger: Logger = Logging.getLogger(DefaultProblemsReportCreator::class.java)
+
+class DefaultProblemsReportCreator(
+    executorFactory: ExecutorFactory?,
+    temporaryFileProvider: TemporaryFileProvider?,
+    internalOptions: InternalOptions?,
+    startParameter: StartParameterInternal,
+    private val failureFactory: FailureFactory,
+    private val buildNameProvider: BuildNameProvider
+) : ProblemReportCreator {
+
+    private val report = CommonReport(executorFactory!!, temporaryFileProvider!!, internalOptions!!, "problem-report")
```

Comment:
🤔  Why not change the types in the constructor to be non-nullable?

In the other place, we use a space-separated string for `"configuration cache report"`. We should probably make sure this is consistent.